### PR TITLE
[ISSUE-34] Wrong slug with umlauts

### PR DESCRIPTION
**Pull request by pedramham**

### CODE REVIEW

**Comment by pedramham** [comment]
Hi @hungmac-sw  When the Dom is updated by changing the language.
I called **getLocaleLanguage()** method. It works But before typing in the input.
I got an error in the console .**localeLanguage** because variable is not set when the page loads. can you check it

`An error was captured in current module: Error: slugify: string argument expected`

**Comment by hungmac-sw** [comment]
@pedramham Because you didn't call `this.getLocaleLanguage()` within `createdComponent`, I guess

File: src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js [hunk 1]
```diff
@@ -80,6 +81,10 @@ Component.register('sas-blog-detail', {
             return this.repositoryFactory.create('sas_blog_entries');
         },
 
+        languageRepository() {
```
Comment (hungmac-sw) [inline src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js:84]: it should be localeRepository

File: src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js [hunk 2]
```diff
@@ -166,6 +171,7 @@ Component.register('sas-blog-detail', {
 
         async changeLanguage() {
             await this.getBlog();
+            await this.slugMaker(slugify(this.slugBlog, {locale:this.localeLanguage, lower: true}));
```
Comment (hungmac-sw) [inline src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js:174]: fix the space
```suggestion
            await this.slugMaker(slugify(this.slugBlog, { locale: this.localeLanguage, lower: true }));
```

File: src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js [hunk 3]
```diff
@@ -225,21 +231,50 @@ Component.register('sas-blog-detail', {
...
+        async getLocaleLanguage() {
+            return this.languageRepository.get(Shopware.Context.api.language.localeId, Shopware.Context.api).then((result) => {
+                this.localeLanguage = result.code.substr(0, result.code.length-3).toLowerCase();
+                return Promise.resolve(this.localeLanguage);
+            });
+        },
+
+        slugMaker(value){
+
+            const valueSlug = slugify(value, {locale:this.localeLanguage, lower: true});
```
Comment (hungmac-sw) [inline src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js:262]: fix the space

File: src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js [hunk 4]
```diff
@@ -225,21 +231,50 @@ Component.register('sas-blog-detail', {
...
+            const criteria = new Criteria();
+            criteria.addFilter(
+                Criteria.equals('slug', valueSlug)
+            );
+
+            this.repository.search(criteria, Shopware.Context.api).then((result) => {
```
Comment (hungmac-sw) [inline src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js:268]: Change this.repository to this.blogRepository

File: src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js [hunk 5]
```diff
@@ -33,6 +33,7 @@ Component.register('sas-blog-detail', {
     data() {
         return {
             blog: null,
+            language: null,
```
Comment (hungmac-sw) [inline src/Resources/app/administration/src/module/blog-module/page/sas-blog-detail/index.js:36]: I don't see it in the code using it

**Pull request merged**